cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -300,61 +302,41 @@ class device_row_comparator {
   std::optional<device_span<null_order const>> const _null_precedence;
 };  // class device_row_comparator
 
+/**
+ * @brief Wraps and interprets the result of templated Comparator that returns a weak_ordering.
+ * Returns true if the weak_ordering matches any of the templated values.
+ *
+ * Note that this should never be used with only `weak_ordering::EQUIVALENT`.
+ * An equality comparator should be used instead for optimal performance.
+ *
+ * @tparam Comparator generic comparator that returns a weak_ordering.
+ * @tparam values weak_ordering parameter pack of orderings to interpret as true
+ */
+template <typename Comparator, weak_ordering... values>
+struct weak_ordering_comparator_impl {
+  __device__ bool operator()(size_type const& lhs, size_type const& rhs)
```

Comment:
We can actually `static_assert` this requirement to be sure it doesn't happen. 

Double check that I'm using the right names here. 
```suggestion
template <typename Comparator, weak_ordering... values>
struct weak_ordering_comparator_impl {
   static_assert( not (sizeof...(values) == 1 and values == weak_ordering::equivalent), "weak_ordering_comparator should not be used for pure equality comparisons. The `row_equality_comparator` should be used instead");
  __device__ bool operator()(size_type const& lhs, size_type const& rhs)
```

Reply:
For reference, the compiler complains about `values == weak_ordering::equivalent` in the context of the static assert. This forces an expansion --> `((weak_ordering::EQUIVALENT == values) && ...)`, which subsequently makes the `sizeof...(values) == 1` check irrelevant.